src/Symfony/Component/Console/Application.php

```diff
@@ -923,6 +925,14 @@ protected function doRunCommand(Command $command, InputInterface $input, OutputI
         $e = null;
 
         try {
+            if ($command instanceof LockableInterface) {
+                $command->getDefinition()->addOption(new InputOption('--skip-lock', '', InputOption::VALUE_NONE, 'Skip checking for and setting a lock'));
+
+                if (false === $input->hasParameterOption(['--skip-lock'], true) && false === $command->lock($command->getName().'_symfony')) {
```

Comment:
i dont think it should pass a $name actually, the null case is a nice extension point for commands to define a name with more variance

Reply:
It's a default value, you can totally override it. See my example implementation of this feature below.

Reply:
i understand, but ignoring a user provided $name cant be the right implementation IMHO.

Id like to do something like `$name = $name ?? 'my-default-command-lock-name'`